Sources/Networking/v2/APIRequestV2.swift

```diff
@@ -36,26 +64,27 @@ public struct APIRequestV2: CustomDebugStringConvertible {
     ///   - cachePolicy: The request cache policy, default is `.useProtocolCachePolicy`
     ///   - responseRequirements: The response requirements
     ///   - allowedQueryReservedCharacters: The characters in this character set will not be URL encoded in the query parameters
-    public init<QueryParams: Collection>(
-        url: URL,
-        method: HTTPRequestMethod = .get,
-        queryItems: QueryParams?,
-        headers: APIRequestV2.HeadersV2? = APIRequestV2.HeadersV2(),
-        body: Data? = nil,
-        timeoutInterval: TimeInterval = 60.0,
-        cachePolicy: URLRequest.CachePolicy? = nil,
-        responseConstraints: [APIResponseConstraints]? = nil,
-        allowedQueryReservedCharacters: CharacterSet? = nil
-    ) where QueryParams.Element == (key: String, value: String) {
+    /// - Note: The init can return nil if the URLComponents fails to parse the provided URL
+    public init?(url: URL,
+                 method: HTTPRequestMethod = .get,
+                 queryItems: QueryItems? = nil,
+                 headers: APIRequestV2.HeadersV2? = APIRequestV2.HeadersV2(),
+                 body: Data? = nil,
+                 timeoutInterval: TimeInterval = 60.0,
+                 retryPolicy: RetryPolicy? = nil,
+                 cachePolicy: URLRequest.CachePolicy? = nil,
+                 responseConstraints: [APIResponseConstraints]? = nil,
+                 allowedQueryReservedCharacters: CharacterSet? = nil) {
 
         self.timeoutInterval = timeoutInterval
         self.responseConstraints = responseConstraints
 
-        let finalURL = if let queryItems {
-            url.appendingParameters(queryItems, allowedReservedCharacters: allowedQueryReservedCharacters)
-        } else {
-            url
+        // Generate URL request
+        guard var urlComps = URLComponents(url: url, resolvingAgainstBaseURL: true) else {
+            return nil
         }
+        urlComps.queryItems = queryItems?.toURLQueryItems(allowedReservedCharacters: allowedQueryReservedCharacters)
```

Comment:
why are we overwriting existing url components with passed query items, is it intended? It seems not intuitive

Reply:
Correct, if a URL already containing query items was passed this would have deleted them, fixed thanks.